unblob/file_utils.py

```diff
@@ -558,3 +570,23 @@ def create_hardlink(self, src: Path, dst: Path):
                     "Not enough privileges to create hardlink to block/char device."
                 )
                 self.record_problem(safe_link.format_report(not_enough_privileges))
+
+    def open_binary(self, path):  # -> io.BufferedRandom
+        # If you want to add a concise explicit return type, be aware of https://github.com/python/typeshed/issues/6077
+        # PyRight manages to get the return type anyway.
+        """Create/open binary file for writing and reading - random access.
+
+        Existing files will not get truncated.
+
+        Expect something equivalent of a io.BufferedRandom object as return value.
+        """
+        logger.debug("create/open binary file for writing", file_path=path)
+        safe_path = self.get_checked_path(path, "open")
+
+        if safe_path:
+            self._ensure_parent_dir(safe_path)
+            mode = "r+b" if safe_path.exists() else "w+b"
+            return safe_path.open(mode)
+
+        # promised of not failing on bad paths:
+        return tempfile.TemporaryFile("w+b")
```

Comment:
not sure why we want to return a temp file instead of failing/raising execption?

Also do we want to allow overwriting existing file by default? I think it should be controllable by the caller

Reply:
The goal of `FileSystem` helper is to ease writing extractors, by removing the burden of dealing with the path traversal problem.

This means, that operations should succeed even if there is a path traversal, but the path traversal attempt is recorded for later reporting.

The problem here is that this operation differs from the rest, as it also returns an object to be used later in the extractor.
If we would raise an exception here or return value None, that would force a check (actually for path traversal) at the extractor side, as without that further operations would fail. So there was a need for a dummy return value, and due to its simplicity the stdlib `TemporaryFile` was used as a safe substitute.

Raising an exception would make this operation potentially fatal for the extractor, while this could be only one bad file of the many to be extracted, so that is not a good solution.

Reply:
I agree, the implicit smartness of opening the file is too much, we should have a control in place to open existing or require new file, maybe also fail on existing as well.

Something like
```py
def open_binary(self, path, non_existing=True, truncate_existing=True):
```
Where `non_existing` would result in `mode="xb+"` (raises error on existing), while `truncate_existing` would result in `mode="wb+"`, while both being False with an existing file would be `mode="rb+"`.